.github/workflows/build-test.yml

```diff
@@ -20,7 +20,8 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # pin@5.0.2
         with:
-          go-version: '1.22.7'
+          go-version: '1.22'
+          check-latest: true
```

Comment:
why not just use `go-version-file: go.mod`

Reply:
This is a good point. I can address that in another PR.

Reply:
https://github.com/helm/helm/pull/13347

Reply:
The `go` directive in the `go.mod` file sets the minimum version of Go that _can_ be used to compile the code. When we build Helm, we want to use the latest patch version on the minor version we are using. This way we get all of the bug and security fixes.

If Helm does use the `go.mod` file (either the `go` or `toolchain` directives) than we will need to update this file with every new Go release to ensure we build against the latest patch version of Go.

I'm not in favor of this extra work. If we went this way we would need to automate the updates to the `go.mod` file.

Reply:
I think with the change I made, it selects the latest version of whatever is specified in the go.mod file.